src/compiler/checker.ts

```diff
@@ -16743,6 +16746,65 @@ namespace ts {
             }
         }
 
+        function markDecoratorMedataDataTypeNodeAsReferenced(node: TypeNode): void {
+            const entityNameOrToken = getEntityNameForDecoratoryMetadata(node);
+            if (entityNameOrToken && isEntityName(entityNameOrToken)) {
+                markEntityNameOrEntityExpressionAsReference(entityNameOrToken);
+            }
+        }
+
+        type voidUndefinedNullOrNeverTypeNode = Token<SyntaxKind.VoidKeyword | SyntaxKind.UndefinedKeyword | SyntaxKind.NullKeyword | SyntaxKind.NeverKeyword>;
+
+        function getEntityNameForDecoratoryMetadata(node: TypeNode): EntityName | voidUndefinedNullOrNeverTypeNode {
+            if (node) {
+                switch (node.kind) {
+                    case SyntaxKind.IntersectionType:
+                    case SyntaxKind.UnionType:
+                        let commonEntityName: EntityName | voidUndefinedNullOrNeverTypeNode;
+                        for (const typeNode of (<UnionOrIntersectionTypeNode>node).types) {
+                            const individualEntityName = getEntityNameForDecoratoryMetadata(typeNode);
+                            if (!individualEntityName) {
+                                // Individual is something like string number 

```

Comment:
is this mean that is each constituent of intersection or union type is a string or number then return undefined?
Can you note where it will get seriazlied?